### DISCUSSION TITLE
Art Credit

### DISCUSSION THREAD

**Discussion by FalcusDev** [RetroAchievements/RAWeb#1096] [Ideas]
I'm opening this request to discuss the potential of being able to add a value to an achievement, where if someone helps to create a badge for the achievement, or set, they can be credited.

What I have in mind would be something like the following images, which can be manually updated by a developer to provide credit to the user who has helped to create badges for games.
Also to be able to give credit on the game page to the user who created a game icon alongside the set authors as well, so that our art team receive appropriate recognition for the great work that they do!

![Art Credit 1](https://user-images.githubusercontent.com/108101508/185709514-f0fb3fd3-4b7a-4915-9417-98153b4369dd.png)
![Art Credit 2](https://user-images.githubusercontent.com/108101508/185709517-018fad3d-3455-48c4-8e06-43664a68b774.png)

**Comment by Jamiras** [comment]
related: #689

**Reply by FalcusDev** [reply]
I'd managed to miss this when searching for related topics. Thank you for highlighting Jamiras.

**Comment by luchaos** [comment]
Permission matrix including permissions specific to artists: https://docs.google.com/spreadsheets/d/17YROKCpqinnyYG3cE8Q1mLDxzOAARCFzvjGltuADZb0/edit#gid=0

**Comment by televandalist** [comment]
I had started trying to track these via multiple spreadsheets with the intent on importing them once we have such a feature. 

The issue with manually updating those is the same issue I had with manually tracking hashes for 3+ years: I don't always know when something changes so the data can quickly become out-of-date. Being able to search for badge updates would help with this until there's a way to add names to the achievement table.

- One thing that's more of an annoyance than an issue is that we have a ridiculous amount of unknowns. Would we default to the achievement author in these cases or leave them blank? I could knock out quite a bit if I get an answer for this.

- How would we handle cases like Trails in the Sky the 3rd where I did the border/background template and ladynadiad finished them with the actual subjects?

**Reply by FalcusDev** [reply]
Manually updating definitely has its own share of issues especially without a way of tracking these updates, but with specific art helper permissions this may not be required if it automatically links to whoever has uploaded them. The primary issue for that though would be if someone did an icon gauntlet and it passed and they don't have an appropriate role, so figuring out an appropriate way to accommodate those types of situations.

- This has made me think of a flaw in my original thoughts, of only listing the credit when there was someone other than the dev who created the art. I'd imagine a large portion of the unknowns are likely created by the dev them self so it would work fine to credit the dev, however leaving it blank if unknown seems a better option, as long as it's possible to update later on when it's known who to credit.
- This is a good point as well. When uploading an icon, potentially the option to include that it was done as a collaboration effort with another user and enter their username so that it displays both something like the following image

The wording could be different, even something like "Art by", I've put last updated by as a placeholder but it depends on how updating it is also handled and record keeping for this.
![Art Credit 3](https://user-images.githubusercontent.com/108101508/185717655-b5194b05-9e35-4511-8243-758ad0aad695.png)

**Reply by luchaos** [reply]
This extends to more than art. An achievements consists of: concept, trigger logic, artwork, writing. All of those should and may be attributed to multiple persons each in the future.

Each change of those assets wants to be tracked individually as they may have been done by a different person. 

Even contributions that have been completely replaced should be attributed.

We end up with version control, git, and how GitHub and similar services handle it, really.
That should be the template.

It will take more than a few lines on the achievement's page - there will have to be a whole section to credit all contributors over time.

**Comment by WilliamFaure** [comment]
![image](https://user-images.githubusercontent.com/103043881/216633401-20050f17-d6ba-4b2b-a410-42d0a6daf213.png)
Posted this one over RA's Discord server. As I'm currently helping with multiple sets, I'm maintaining my own spreadsheet to keep track of my contributions. This is how it's looking, basically - please consider this screenshot as **site feedback**, and possible help for improving it when it comes to properly crediting achievement authors. Thank you.

**Comment by wescopeland** [comment]
This is now implemented in the React game page live in production.
